rpc/gateway/service.proto

```diff
@@ -0,0 +1,299 @@
+syntax = "proto3";
+
+package razorpay.gateway;
+
+// import "google/protobuf/struct.proto";
+
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
+  schemes: [HTTP, HTTPS];
+  base_path: "/twirp/";
+  security: {
+    security_requirement :{
+      key: "ApiKeyAuth";
+      value: {};
+    }
+  }
+  security_definitions: {
+    security: {
+      key: "ApiKeyAuth";
+      value: {
+        type: TYPE_API_KEY;
+        in: IN_HEADER;
+        name: "X-Auth-Key";
+      };
+    };
+    // security: {
+    //   key: "BasicAuth";
+    //   value: {
+    //       type: TYPE_BASIC;
+    //   }
+    // };
+  }
+};
+
+option go_package = "./rpc/gateway;gatewayv1";
+
+message Empty {}
+
+service HealthCheckAPI {
+    // Check returns HealthCheckResponse for the requested service.
+    rpc Check(HealthCheckRequest) returns (HealthCheckResponse);
+}
+
+message HealthCheckRequest {
+    string service = 1;
+}
+
+message HealthCheckResponse {
+    // ServingStatus denotes the status of the service.
+    enum ServingStatus {
+        SERVING_STATUS_INVALID = 0;
```

Comment:
What is the difference between `SERVING_STATUS_INVALID`, `SERVING_STATUS_UNKNOWN` and `SERVING_STATUS_NOT_SERVING`. I can see that when the health check returns not `ok`, the code will return `SERVING_STATUS_NOT_SERVING`. When will the other two be used?